src/devices/logitech_g535.c

```diff
@@ -4,14 +4,79 @@
 
 #include <math.h>
 #include <string.h>
-
-#define MSG_SIZE 20
+#include <unistd.h>
 
 static struct device device_g535;
 
 static const uint16_t PRODUCT_ID = 0x0ac4;
 
+// Based on manual measurements so the discharge curve used to generate these values aren't always
+// right, but it's good enough.
+// Based on the following measured values on a brand new headset (voltage, percentage) :
+// - 4175, 100
```

Comment:
This takes a lot of space but I think it's useful information if somebody else wants to implement the discharge curve or battery status in another way.
Maybe put it in another file somewhere else, a CSV file for example ? :shrug:

Reply:
I think instead of a polynomial function its better to have a spline. A spline is much simpler and is more transparent in defining it. 

here is sample code for  that: https://github.com/Sapd/HeadsetControl/pull/197#issuecomment-1006922692

it btw. does not need to the that accurate, as every battery will be different anyway, even of the same headset

Reply:
I've implemented a utility function to use splines to compute the battery level :+1: 
I'd still keep all the measured values just for documentation